haystack_experimental/components/tools/openai/function_caller.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+from typing import List
+
+from haystack import component
+from haystack.dataclasses import ChatMessage
+
+
+@component
+class OpenAIFunctionCaller:
+    """
+    The OpenAIFunctionCaller expects a list of ChatMessages and if there is a tool call with a function name and arguments, it runs the function and returns the
+    result as a ChatMessage from role = 'function'
+    """
+
+    def __init__(self, available_functions):
+        """
+        Initialize the OpenAIFunctionCaller component.
+        :param available_functions: A dictionary of available functions. This dictionary expects key value pairs of function name, and the function itslelf. For example {"weather_function": weather_function}
+        """
+        self.available_functions = available_functions
```

Comment:
This won't work for serialization, you have to call https://github.com/deepset-ai/haystack/blob/main/haystack/utils/callable_serialization.py#L12 and store the resulting string

Reply:
@masci I've just made a commit for this, can you tell me if this is what you'd expect?